dbms/src/Columns/ColumnArray.cpp

```diff
@@ -178,11 +178,16 @@ StringRef ColumnArray::serializeValueIntoArena(size_t n, Arena & arena, char con
     char * pos = arena.allocContinue(sizeof(array_size), begin);
     memcpy(pos, &array_size, sizeof(array_size));
 
-    size_t values_size = 0;
+    StringRef res(pos, sizeof(array_size));
+
     for (size_t i = 0; i < array_size; ++i)
-        values_size += getData().serializeValueIntoArena(offset + i, arena, begin).size;
+    {
+        auto value_ref = getData().serializeValueIntoArena(offset + i, arena, begin);
+        res.data = value_ref.data - res.size;
+        res.size += value_ref.size;
```

Comment:
Can we just remember the offset from the start of chunk (begin) instead of these calculations?

Reply:
Yes.
But begin can be nullptr at begin, and can change value after `serializeValueIntoArena` several times. Also we don't know offset from begin beforehand. 
I think current code is better.
(You may try to rewrite it)

Reply:
I'd just add a method to Arena that returns current offset from the start of chunk. We're relying on it to track this offset, anyway. But we don't really have to do this in this bugfix, so let's keep your version.